unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/GeneratedCode/RendererProtocolService.gen.cs

```diff
@@ -0,0 +1,44 @@
+// AUTOGENERATED, DO NOT EDIT
+// Type definitions for server implementations of ports.
+// package: 
+// file: RendererProtocol.proto
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using Google.Protobuf;
+using rpc_csharp.protocol;
+using rpc_csharp;
+
+public abstract class CRDTService<Context>
+{
+    public const string ServiceName = "CRDTService";
+
+    public delegate UniTask<CRDTResponse> SendCrdt(CRDTManyMessages request, Context context);
```

Comment:
I'd assume we need a Cancellation Token for this.

Reply:
This is called by kernel and it won't try to cancel the task.
So any cancellation needed should be handled by the implementation of those delegates.
a quick example might be:
```
UniTask<RpcResponse> SomeProceduralCall(Request someRequest, Context context){
     try{
          var result = await SomeCancellableTask(someRequest, context);
          return responseTaskOK;
     } catch (Exception e){
          return responseTaskException;
     }
}
```